### DISCUSSION TITLE
how to actually use laragon php versioning

### DISCUSSION THREAD

**Discussion by gustiRanda** [leokhoa/laragon#894] [Q&A] [answered]
so I have installed laragon and configure the php, node, and npm, and when I check the version from laragon terminal, it shows their versions correctly, then I created a laravel project and try to instal tailwind which required a npm, it gives me an error message

'npm : The term 'npm' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the 
name, or if a path was included, verify that the path is correct and try again.'

then I tried to check the node and npm version via windows terminal, and it gives me an error 

''node' is not recognized as an internal or external command,'

it's like the php, node, and npm only get installed in laragon own environment

I need to manually configure the php, node, and npm path manually in the environment variables on my laptop
thats like destroy the point of laragon right? can someone help me fix this?

**Comment by Pen-y-Fan** [comment, marked as answer]
To add Laragon to your paths (see [laragon.org > docs > isolated](https://laragon.org/docs/isolated.html) and my blog  [How to set up Laragon on a new Windows computer (part 6) - Add to path](https://pen-y-fan.github.io/2023/01/05/how-to-set-up-laragon-on-a-new-windows-computer-part-6/)), just click **Menu > Tools > PATH environment variable > Add Laragon to Path**

You then need to **Log off** and back **on** for the path to be added to all terminals (CMD/ Shell / Terminal etc.)

One point to note: if you change PHP version, you need to run **Remove Laragon from Path**, log off and back on. Then **Add Laragon to Path** log off and back on!

The new PHP version will then be available in all terminals.

**Reply by gustiRanda** [reply]
I dont understand the log off and on part, can you please expalinit?

**Reply by Pen-y-Fan** [reply]
When using Windows 10 the changes to PATH would take effect at log on. 

I believe Windows 11 PATH changes takes effect when changed. Any open terminals will need to be closed, by logging off and back on the PATH will be used when terminals are opened.

The best thing to do is try it. Set the PATH using Menu > Tools > PATH environment variable > Add Laragon to Path, then open a terminal and check npm, node etc.  Confirm it works for your computer.

The [terminal built into Laragon](https://laragon.org/docs/screenshots#Terminal) will autocratically have the environment variables set.

**Reply by gustiRanda** [reply]
thank you, you solved my problem
